src/api.rs

```diff
@@ -1407,45 +1433,53 @@ impl<T: Pixel> ContextInner<T> {
   }
 
   fn compute_lookahead_data(&mut self) {
-    // Process all new frames that we have received.
-    let frames_to_process = self
-      .frame_q
-      .iter()
-      .filter_map(|(&input_frameno, frame)| {
-        if input_frameno >= self.next_lookahead_frame {
-          frame.clone().map(|frame| (input_frameno, frame))
-        } else {
-          None
-        }
-      })
-      .collect::<Vec<_>>();
-
-    for &(input_frameno, ref frame) in frames_to_process.iter() {
-      if self.is_key_frame(input_frameno, frame) {
-        self.keyframes.insert(input_frameno);
+    let lookahead_frames = self
+        .frame_q
+        .iter()
+        .filter_map(|(&input_frameno, frame)| {
+          if input_frameno >= self.next_lookahead_frame {
+            frame.clone()
+          } else {
+            None
+          }
+        })
+        .take(self.inter_cfg.keyframe_lookahead_distance() as usize + 1)
+        .collect::<Vec<_>>();
+    let mut lookahead_idx = 0;
+
+    while self.lookahead_ready() {
+      // Process the next unprocessed frame
+      // Start by getting that frame and all frames after it in the queue
+      let lookahead_frames = &lookahead_frames[lookahead_idx..];
```

Comment:
This looks suspicious and confusing. Please do not reslice `lookahead_frames` this way.

Reply:
@YaLTeR had asked me to make this change for performance reasons, instead of running over the iterator inside the loop. I agree it would be clearer what is happening without this reslice.

Reply:
Changing the local variable name would be enough to avoid the confusion, possibly we can rework the loop later.